src/iccpd/src/mlacp_sync_update.c

```diff
@@ -0,0 +1,659 @@
+    {
+        mac_msg = (struct MACMsg*)msg->buf;
+
+        /*Same MAC is exist in local switch*/
+        if (strcmp(mac_msg->mac_str, tlv->mac_str) == 0 && mac_msg->vid == ntohs(tlv->vid))
+        {
+            if (tlv->type == MAC_SYNC_ADD)
+            {
+                mac_msg->age_flag &= ~MAC_AGE_PEER;
+                ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ADD, Remove peer age flag:%d ifname  %s, add %s vlan-id %d, op_type %d",
+                                mac_msg->age_flag, mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+
+                /*mac_msg->fdb_type = tlv->fdb_type;*/
+                /*The port ifname is different to the local item*/
+                if (strcmp(mac_msg->ifname, tlv->ifname) != 0 || strcmp(mac_msg->origin_ifname, tlv->ifname) != 0)
+                {
+                    if (mac_msg->fdb_type != MAC_TYPE_STATIC)
+                    {
+                        /*Update local item*/
+                        memcpy(&mac_msg->ifname, tlv->ifname, MAX_L_PORT_NAME);
+                        memcpy(&mac_msg->origin_ifname, tlv->ifname, MAX_L_PORT_NAME);
+                    }
+
+                    /*If the MAC is learned from orphan port, or from MCLAG port but the local port is down*/
+                    if (from_mclag_intf == 0 || (local_if->state == PORT_STATE_DOWN && strcmp(mac_msg->ifname, csm->peer_itf_name) != 0))
+                    {
+                        /*Redirect the mac to peer-link*/
+                        memcpy(&mac_msg->ifname, csm->peer_itf_name, IFNAMSIZ);
+
+                        /*Set MAC_AGE_LOCAL flag*/
+                        mac_msg->age_flag = set_mac_local_age_flag(csm, mac_msg, 1);
+
+                        /*Send mac add message to mclagsyncd*/
+                        add_mac_to_chip(mac_msg, MAC_TYPE_DYNAMIC);
+                    }
+                }
+
+                #if 0
+                mac_msg->op_type = MAC_SYNC_ACK;
+                if (iccp_csm_init_msg(&msg_send, (char*)mac_msg, sizeof(struct MACMsg)) == 0)
+                {
+                    /*Reply mac ack message to peer, peer will clean MAC_AGE_PEER flag*/
+                    TAILQ_INSERT_TAIL(&(MLACP(csm).mac_msg_list), msg_send, tail);
+                    ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ADD, MAC-msg-list enqueue: %s, add %s vlan-id %d, op_type %d",
+                                    mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+                }
+                #endif
+            }
+            else if (tlv->type == MAC_SYNC_ACK)
+            {
+                /*Clean the MAC_AGE_PEER flag*/
+                mac_msg->age_flag &= ~MAC_AGE_PEER;
+                ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ACK, Remove peer age flag:%d ifname  %s, add %s vlan-id %d, op_type %d",
+                                mac_msg->age_flag, mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+            }
```

Comment:
It looks the code to send the MAC_SYNC_ACK is being used, this unused code can be commented. 
Can someone please clarify what is the original intention of having the MAC_SYNC_ACK, is this to handle  MAC out of sync scenarios between McLAG peer nodes .? Also is there any plan to make the changes active .?

Reply:
OK, we will delete these unused codes. Yes, the original intention of MAC_SYNC_ACK is to prevent MAC out of sync. If anyone find any scenario that MAC out of sync in current implementation, we will refine it.